### PR TITLE
Create base from UI

### PR DESCRIPTION
**Pull request by tomeronen**
Starting to build the UI step by step. This is the first step contains basic setup discussions about the libraries to use, and building the code as part of the martin build and serving it from the binariy :
- js library - **_React with TS_**: Both kepler.gl and Maputnik are written in react. Which can make bringing them in some way to this UI easier.
- bundler - **_vite_**.
- styling - **_styled-components_** my personal favourite (and used by kepler.gl so we will be able to easily learn from what they have done)

### CODE REVIEW

File: martin-ui/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# React + TypeScript + Vite
```
Comment (nyurik) [inline martin-ui/README.md:1]: this needs updating - I would prefer not to check in unrelated boilerplate readme

File: martin-ui/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import martinCover from './assets/martin-cover.png'
```
Comment (nyurik) [inline martin-ui/src/App.tsx:1]: this image is somewhat incorrect - martin is not only focused on large databases... Not sure what to say there - maybe even leave it at empty-ish, or use the martin's main documentation string?

File: martin/src/srv/server.rs [hunk 3]
```diff
@@ -96,6 +104,15 @@ pub fn router(cfg: &mut web::ServiceConfig) {
 
     #[cfg(feature = "fonts")]
     cfg.service(crate::srv::fonts::get_font);
+
+    #[cfg(feature = "webui")]
+    cfg.service(actix_web_static_files::ResourceFiles::new(
```
Comment (nyurik) [inline martin/src/srv/server.rs:113]: there is a subtle bug here - compiled files go into `/assets/...` - but the word `assets` could also be the name of a tile source. Is there a way we can make the root `/index.html` use `/_/assets/...` path instead?  The `_` is a reserved keyword, so we should put everything non-stable there.

**Review by nyurik — COMMENTED** [review]
minor nits, and we need some sort of a basic "smoke test" in CI - something that will check if webui is actually loading properly - perhaps add a simple test to curl `/` and some asset (whose name does not change)?

**Comment by nyurik** [comment]
@tomeronen i merged in the changes from the main branch. Let me know if you need any help with this

**Comment by nyurik** [comment]
FYI, Maputnik has been moved to [maplibre](https://github.com/maplibre/maputnik), and so is the [maplibre-gl-inspect](https://github.com/maplibre/maplibre-gl-inspect), with lots of work being done on them.  Any updates on this PR?

**Comment by paigewilliams** [comment]
@tomeronen @nyurik I'd be happy to help out on this if you'd like!

**Comment by nyurik** [comment]
> @tomeronen @nyurik I'd be happy to help out on this if you'd like! 

Any help is welcome! Thx 😊

**Comment by paigewilliams** [comment]
I made some changes and submitted them here #1395

**Comment by CommanderStorm** [comment]
(sorry to bother: given that @paigewilliams' PR https://github.com/maplibre/martin/pull/1395 is merged, I think this PR should is redundant ^^)

**Comment by nyurik** [comment]
I'll check shortly, ie if it has any extra code that might still be useful?

**Comment by nyurik** [comment]
All changes from these PRs were done in #1395. Thx for all the hard work that lead to adding static hosting!

**Pull request closed without merging**